docs/specification.md

```diff
@@ -0,0 +1,1280 @@
+  ( string         :tag
+  , option address :configApi
+  )
+
+permissionsDescriptorParam
+  ( selfTransferPolicy            :self
+  , operatorTransferPolicy        :receiver
+  , ownerTransferPolicy           :operator
+  , ownerTransferPolicy           :sender
+  , option customPermissionPolicy :custom
+  )
+
+permissionsDescriptor = contract permissionsDescriptorParam
+```
+
+Parameter (in Michelson)
+```
+(contract %permissionsDescriptor
+  (pair
+    (or %self
+      (unit %selfTransferPermitted)
+      (unit %selfTransferDenied)
+    )
+    (pair
+      (or %operator
+        (unit %operatorTransferPermitted)
+        (unit %operatorTransferDenied)
+      )
+      (pair
+        (or %receiver
+          (unit %ownerNoOp)
+          (or
+            (unit %optionalOwnerHook)
+            (unit %requiredOwnerHook)
+        ))
+        (pair
+          (or %sender
+            (unit %ownerNoOp)
+            (or
+              (unit %optionalOwnerHook)
+              (unit %requiredOwnerHook)
+          ))
+        (option %custom
+          (pair
+            (string %tag)
+            (option %configApi address)
+          )
+        )
+  ))))
+)
+```
+
+- Get the descriptor of the transfer permission policy. This allows
+  for external contracts to discover an FA2 contract's permission
+  policy as well as to configure it. For additional information
+  please refer to FA2 token contract specification.
+
+- Some of the permission options require an additional configuration
+  API which can be implemented either within FA2 token contract
+  itself or in a separate contract.
+
+Each minter is separated into 2 types of addresses: `operator` and
+`owner`. `operator` is a Tezos address that initiates token tranfser
+operation on behalf of the `owner` that actually holds tokens.
+
+**updateOperators**
+
+Types
+```
+tokenId = nat
+
+operatorTokens =
+  | AllTokens
+  | SomeTokens (set tokenId)
+
+operatorParam =
+  ( address        :owner
+  , address        :operator
+  , operatorTokens :tokens
+  )
+
+updateOperatorParam =
+  | AddOperator    operatorParam
+  | RemoveOperator operatorParam
+
+updateOperators = list updateOperatorParam
+```
+
+Parameter (in Michelson)
+```
+(list %updateOperators
+  (or
+    (pair %addOperator
+      (address %owner)
+      (pair
+        (address %operator)
+        (or %tokens
+          (unit %allTokens)
+          (set %someTokens nat)
+        )
+    ))
+    (pair %removeOperator
+      (address %owner)
+      (pair
+        (address %operator)
+        (or %tokens
+          (unit %allTokens)
+          (set %someTokens nat)
+        )
+    ))
+  )
+)
+```
+
+- Add or Remove token operators for the specified owners and token
+  types.
+
+- If two different commands in the list add and remove an operator
+  for the same token type, then the last one must take effect.
+
+- It's possible to update an operator for some specific tokens, or
+  to all tokens (TODO: discuss).
+
+- Sender must be minter.
```

Comment:
At first I thought that it's be better to separate roles to minters so that we can assure that wallet has some balance before transferring. On a second thought it can be implemented easier just failing with an error if it does not have enough coins to transfer.